arm7/lib/src/SND_seq.c

```diff
@@ -0,0 +1,1323 @@
+    {
+        struct SNDTrack *trk = PlayerGetTrack(player, i);
+        if (trk)
+        {
+            TrackUpdateChannel(trk, player, 1);
+        }
+    }
+}
+
+static void TrackPlayNote(
+    struct SNDTrack *track, struct SNDPlayer *player, int midiKey, int velocity, int length)
+{
+    struct SNDExChannel *chn = NULL;
+
+    if (track->flags.tie)
+    {
+        chn = track->channelLLHead;
+        if (chn)
+        {
+            chn->midiKey = (u8)midiKey;
+            chn->velocity = (u8)velocity;
+        }
+    }
+
+    if (chn == NULL)
+    {
+        struct SNDInstData inst;
+        if (!SND_ReadInstData(player->bank, track->program, midiKey, &inst))
+            return;
+
+        u32 allowedChannels;
+
+        switch (inst.type)
+        {
+            case SND_INST_PCM:
+            case SND_INST_DIRECTPCM:
+                // all channels support PCM
+                allowedChannels = 0xFFFF;
+                break;
+            case SND_INST_PSG:
+                // only channels 8, 9, 10, 11, 12, 13 support PSG
+                allowedChannels = 0x3F00;
+                break;
+            case SND_INST_NOISE:
+                // only channels 14 and 15 support noise
+                allowedChannels = 0xC000;
+                break;
+            default:
+                return;
+        }
+
+        allowedChannels &= track->channelMask;
+
+        chn = SND_AllocExChannel(allowedChannels,
+            player->prio + track->priority,
+            track->flags.channelMask,
+            ChannelCallback,
+            track);
+        if (chn == NULL)
+            return;
+
+        if (!SND_NoteOn(
+                chn, midiKey, velocity, track->flags.tie ? -1 : length, player->bank, &inst))
+        {
+            chn->priority = 0;
+            SND_FreeExChannel(chn);
+            return;
+        }
+
+        chn->channelLLNext = track->channelLLHead;
+        track->channelLLHead = chn;
+    }
+
+    if (track->envAttack != 0xFF)
+        SND_SetExChannelAttack(chn, track->envAttack);
+
+    if (track->envDecay != 0xFF)
+        SND_SetExChannelDecay(chn, track->envDecay);
+
+    if (track->envSustain != 0xFF)
+        SND_SetExChannelSustain(chn, track->envSustain);
+
+    if (track->envRelease != 0xFF)
+        SND_SetExChannelRelease(chn, track->envRelease);
+
+    chn->sweepPitch = track->sweepPitch;
+    if (track->flags.portamento)
+        chn->sweepPitch += (short)((track->portamentoKey - midiKey) << 6);
+
+    if (track->portamentoTime == 0)
+    {
+        chn->sweepLength = length;
+        chn->flags.autoSweep = FALSE;
+    }
+    else
+    {
+        int swp = track->portamentoTime * track->portamentoTime;
+        swp *= chn->sweepPitch < 0 ? -chn->sweepPitch : chn->sweepPitch;
+        swp >>= 11;
+        chn->sweepLength = swp;
+    }
+
+    chn->sweepCounter = 0;
+}
+
+static int TrackStepTicks(
+    struct SNDTrack *track, struct SNDPlayer *player, int trackIdx, u32 playNotes)
+{
+    (void)trackIdx;
```

Comment:
pragma unused?

Reply:
What do you mean?

Reply:
I mean, rather than doing
(void)trackIdx
can you do
#pragma unused (trackIdx)

Reply:
I would disagree because doing `(void)var` is more standard compliant and it's exactly how you are supposed to do it.

Reply:
I guess, though in other places in the code we do use #pragma unused

Reply:
Then I'd suggest to change the code to adapt the standard compliant way.